src/rospack.cpp

```diff
@@ -179,11 +179,8 @@ class Stackage
       assert(manifest_loaded_);
       // get name from package.xml instead of folder name
       tinyxml2::XMLElement* root = get_manifest_root(this);
-      for(tinyxml2::XMLElement* el = root->FirstChildElement("name"); el; el = el->NextSiblingElement("name"))
-      {
-        name_ = el->GetText();
-        break;
-      }
+      if(root->FirstChildElement("name"))
+        name_ = root->GetText();
```

Comment:
Wasn't the former logic setting `name_` to `root->FirstChildElement("name")->GetText()`?

Reply:
You are right. Sorry missed it